content/en/apps/guides/hosting/4.x/self-hosting-k3s-multinode.md

```diff
@@ -0,0 +1,415 @@
+    3. Click Yes to confirm the upgrade.
+    4. Select a compatibility and click OK.
+
+3. Add VMware Paravirtual SCSI storage controller to the VM
+    1. In the vSphere Client, right-click the VM and select Edit Settings.
+    2. On the Virtual Hardware tab, click the Add New Device button.
+    3. Select SCSI Controller from the drop-down menu.
+    4. Expand New SCSI controller and from the Change Type menu, select VMware Paravirtual.
+    5. Click OK.
+
+
+### Identify vSphere Provider IDs, Node IDs, and datacenter name
+
+Bootstrap parameters for k3s on VMware require UUID identification of each node that will join the cluster. 
+
+For each of the provisioned VMs, you can navigate to the VM in vCenter interface and retrieve the UUID.
+
+Another method is to make the following calls to vCenter Server API. You may have a VPN that you connect to first before being able to access your vCenter GUI. These commands should be run from the same network that allows that access.
+
+When running the commands below, be sure to replace the placeholders with your own values:
+* `<vCenter_IP>`
+* `<USERNAME>` 
+* `<PASSWORD>`
+
+And any others as well!
+
+* Get an authentication-token:
+      
+      curl -k -X POST https://<vCenter_IP>/rest/com/vmware/cis/session -u '<USERNAME>:<PASSWORD>'
+ID=<UUID_FROM_vCENTER>
+      
+
+* List all your VMs and identify the VM-number that was provisioned earlier:
+      
+      curl -k -X GET -H "vmware-api-session-id: $ID" https://<vCenter_IP>/api/vcenter/vm
+      
+* Retrieve your instance_uuid by first making a `curl` call:
+      
+      curl -k -X GET -H "vmware-api-session-id: $ID" https://<vCenter_IP>/api/vcenter/vm/vm-<number>
+      
+      
+* Inside the JSON response of the `curl` call get the,  `instance_uuid`, in this case it's `215cc603-e8da-5iua-3333-a2402c05121`, but yours will be different:
+      
+      "identity":{"name":"k3s_worker_node_4","instance_uuid":"215cc603-e8da-5iua-3333-a2402c05121"
+      
+
+* Retrieve your datacenter name, to be used in configuration files for VMware CSI and CPI
+      
+      curl -k -X GET -H "vmware-api-session-id: $ID" https://<vCenter_IP>/rest/vcenter/datacenter
+      
+You will want to save the "name" of your datacenter.
+
+* Retrieve your cluster-id, to be used in config file for VMware CSI
+      
+      curl -k -X GET -H "vmware-api-session-id: $ID" https://<vCenter IP>/api/vcenter/cluster
+      
+
+You can also use the [govc cli tool](https://github.com/vmware/govmomi/blob/main/govc/README.md#binaries) to retrieve this information:
+   
+   export GOVC_INSECURE=1
+   export GOVC_URL='https://<USERNAME>:<PASSWORD>@<vCenter_IP>
+   
+   govc ls /
+        <datacenter-name>/vm \
+        <datacenter-name>/network \
+        <datacenter-name>/host \
+        <datacenter-name>/datastore 
+   
+   #To retrieve all Node VMs
+   govc ls /<datacenter-name>/vm \
+       <datacenter-name>/vm/<vm-name1> \
```

Comment:
The tags disappears for me in the preview, does it look correctly in when rendered by the static site generator?

Minor: I'd prefer to wrap all code lines in backticks (replace `shell` with the language that is most appropriate or leave blank)
````
```shell
code goes here...
```
````
 or for shorter sections
`` `code goes here...` ``